sdks/go/pkg/beam/io/filesystem/gcs/gcs_test.go

```diff
@@ -36,3 +40,238 @@ func TestGCS_FilesystemNew(t *testing.T) {
 		t.Errorf("c.Close() = %v, want nil", err)
 	}
 }
+
+func TestGCS_direct(t *testing.T) {
+	ctx := context.Background()
+	dirPath := "gs://beamgogcsfilesystemtest"
+	filePath := dirPath + "/file.txt"
+
+	server := fakestorage.NewServer([]fakestorage.Object{
+		{
+			ObjectAttrs: fakestorage.ObjectAttrs{
+				BucketName: "beamgogcsfilesystemtest",
+				Name:       "stub",
+			},
+			Content: []byte(""),
+		},
+	})
+	defer server.Stop()
```

Comment:
Prefer using t.Cleanup() over defer, which lets you put the `server.Stop()` call into the helper function. See https://google.github.io/styleguide/go/decisions#test-helpers